lib/saml/settings_service.rb

```diff
@@ -1,28 +1,43 @@
 # frozen_string_literal: true
 require 'memoist'
+require 'sentry_logging'
 
 module SAML
   # This class is responsible for putting together a complete ruby-saml
   # SETTINGS object, meaning, our static SP settings + the IDP settings
   # which must be fetched once and only once via IDP metadata.
   class SettingsService
     class << self
+      include SentryLogging
       extend Memoist
 
       METADATA_RETRIES = 3
       OPEN_TIMEOUT = 2
       TIMEOUT = 15
 
       def saml_settings
+        if @fetch_attempted.nil? || metadata_successfully_retrieved?
+          merged_saml_settings
+        else
+          # retrieve was attempted but failed, blow away memoized data and try again
+          merged_saml_settings(true)
```

Comment:
@omgitsbillryan here's what I was thinking. This would be crappy if the call to `OneLogin::RubySaml::IdpMetadataParser.new.parse` is itself slow. I'm assuming the `connection.get` is the slow part. The main thing I'm trying to achieve here is keeping the retry logic all in one method.

```ruby
module SAML
  class SettingsService
    class << self
      def saml_settings
        merged_saml_settings
      end

      def merged_saml_settings
        begin
          OneLogin::RubySaml::IdpMetadataParser.new.parse(metadata, settings: settings)
        rescue
          # Your rescue logic
        end
      end

      def metadata
        if (we have successfully pulled the metadata)
          it was stored in an instance variable, so return that
        else
          try to pull the metadata with your retry logic, and store it in an instance variable if successful
        end
    end
  end
end
```